framework/src/st_things/st_things_request_handler.c

```diff
@@ -406,7 +406,7 @@ static bool get_supported_properties(const char *res_type, const char *res_uri,
 				// If this prop is already added, then ignore it and decrement the total count by 1.
 				bool exist = false;
 				for (int i = 0; i < cur_index; i++) {
-					if (0 == strncmp(((*properties)[i])->key, prop->key, strlen(prop->key))) {
+					if (0 == strncmp(((*properties)[i])->key, prop->key, sizeof(prop->key))) {
 						THINGS_LOG_V(THINGS_DEBUG, TAG, "Property(%s) is already added in the request message.", prop->key);
```

Comment:
The purpose of replacing strlen to strncmp is to prevent data-abort for corrupted data.

1. modification of strlen+1 if fine.
--> this maybe answer to @sunghan-chang 

@HJ,
2. Of course sizeof is defined on build time, and fater than strlen. but we'll have limiation for maintenance

e.g) If there's chance to change char key to char *key for saving memory on refactoring. it's hard to maintain on each line
typedef struct things_attribute_info_s {
	char key[MAX_KEY_LENGTH];
	char spec[MAX_SPEC_LENGTH];
	int type;
	bool mandatory;
	int rw;
} things_attribute_info_s;
Thus, if you want to use const value, it's better to use the definition MAX_KEY_LENGTH, this will make sense.

doing strlen for undefined memory does not help for the security